plugins/services/src/js/utils/TaskUtil.js

```diff
@@ -112,6 +113,71 @@ const TaskUtil = {
     return ipAddresses.map(function(item) {
       return item.ip_address;
     });
+  },
+
+  getRegionName(task) {
+    const node = this.getNode(task);
+    if (!node) {
+      return "(Local)";
+    }
+
+    const masterNode = CompositeState.getMasterNode();
```

Comment:
Not blocking as this is following existing "patterns" but we should try hard to write _pure_ utils that don't access any kind of data "store" directly.